command/do_transition_test.go

```diff
@@ -0,0 +1,98 @@
+package command
+
+import (
+	"encoding/json"
+	"github.com/ExpediaGroup/flyte-jira/client"
+	"github.com/HotelsDotCom/flyte-client/flyte"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+	"testing"
+)
+
+type (
+	mockRequestBody struct {
+		TransitionId string `json:"id"`
+	}
+
+	DoTransitionRequest struct {
+		Transition mockRequestBody `json:"transition"`
+	}
+)
+
+func createMockSendRequest(issueId, transitionId string) func(request *http.Request) (int, error) {
+	return func(request *http.Request) (int, error) {
+		b, err := ioutil.ReadAll(request.Body)
+		if err != nil {
+			return 400, err
+		}
+
+		body := &DoTransitionRequest{}
+
+		if err := json.Unmarshal(b, &body); err != nil {
+			return 400, err
+		}
+
+		if body.Transition.TransitionId != transitionId {
+			return 500, err
+		}
+
+		if issueId != "DEVEX-123" {
+			return 404, err
+		}
+
+		return 204, nil
+	}
+}
+func TestDoTransitionAsExpected(t *testing.T) {
+	client.SendRequestWithoutResp = createMockSendRequest("DEVEX-123", "881")
```

Comment:
Not super keen on the way the testing is setup across the project. However it is already used and I guess we have to live with it.

Ideally I would like to see usage of `net/http/httptest` to create test server removing the need to change `client.SendRequest....`

However if we follow approach taken through out there are still some things which should be adhere to.

In this particular case we are altering global state and never reset it. Test should leave env in the same state as it found it. General practice is to take copy of the var we are altering, defer func to set it back to this prev value and then set value to something new as you are doing in this step.

Another thing which is missing from the test we are not asserting request path and method basically this in the client `path := fmt.Sprintf("/rest/api/2/issue/%s/transitions", issueId)` is not getting tested and it is kind of important thing